### DISCUSSION TITLE
[Feature Request] Allow to extend autocomplete's inputwrapper

### DISCUSSION THREAD

**Discussion by simPod** [heroui-inc/heroui#4717] [Feature requests]
### Is your feature request related to a problem? Please describe.

currently there's no way to extend an autocomplete's variant with input's style override.

### Describe the solution you'd like

```ts
const Extended = extendVariants(NextUIAutocomplete, {
  variants: {
    variant: {
      faded: {
        inputWrapper: [
          'bg-default-900',
          // 'border-medium',
          // 'border-default-300',
          // 'data-[hover=true]:border-default-500',
        ],
      },
    },
  },
});
```

### Describe alternatives you've considered

```tsx
  <Extended
    variant="faded"
    inputProps={{
      classNames: {
        inputWrapper: [
```

but that feels like a hack

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1544/[feature-request]-allow-to-extend-autocompletes-inputwrapper">ENG-1544 [Feature Request] Allow to extend autocomplete's inputwrapper</a></p>

**Comment by simPod** [comment]
idk, bot got mad https://github.com/nextui-org/nextui/issues/2975, dupe

**Comment by mikaeledgren** [comment]
+1 on this request!

**Comment by MrLexisDev** [comment]
I there any news or solution?

**Comment by dinbtechit** [comment]
is there any solution for this?

**Comment by wingkwong** [comment]
inputProps.classNames.inputWrapper

**Reply by dinbtechit** [reply]
Hmm.. how can I use the inputprops for  auto complete?

**Reply by wingkwong** [reply]
see https://www.heroui.com/docs/components/autocomplete#customizing-the-autocomplete

**Reply by dinbtechit** [reply]
Yep, I already saw that but that that doesn't give me enough to customize the input wrapper

**Reply by wingkwong** [reply]
mind to elaborate more on your case?

**Reply by dinbtechit** [reply]
Actually, the `inputProp` worked! I should read the documentation more carefully. Thank you for pointing me in the right direction.

```jsx
<Autocomplete>
...
 inputProps={{
        classNames: {
          input: "mt-2",
          inputWrapper: "h-20",
        },
      }}
</Autocomplete>
```
